### PR TITLE
Strip original contig splitting approach and preanno orf attempt3

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by jasmezz — COMMENTED** [review]
Some docs comments

File: conf/modules.config [hunk 1]
```diff
@@ -287,6 +248,7 @@ process {
     }
 
     withName: FARGENE {
+        tag = {"${meta.id}|${hmm_model}"}
```
[the remark this answers is not in the document]
Comment (jfy133) [reply conf/modules.config:251]: Yes, keeps it more compact

File: conf/test_taxonomy.config [hunk 2]
```diff
@@ -36,3 +36,9 @@ params {
     run_bgc_screening       = true
     bgc_skip_deepbgc        = true
 }
+
+process {
+    withName: MMSEQS_DATABASES {
+        memory = '12.GB'
```
Comment (jfy133) [inline conf/test_taxonomy.config:42]:
```suggestion
        memory = '14.GB'
```

File: workflows/funcscan.nf [hunk 3]
```diff
@@ -80,42 +66,69 @@ workflow FUNCSCAN {
...
+                                .mix( ch_input_prep.uncompressed )
+                                .groupTuple()
+                                .map{
+                                    meta, files ->
+                                        def fasta_found   = files.find{it.toString().tokenize('.').last().matches('fasta|fas|fna|fa')}
+                                        def faa_found     = files.find{it.toString().endsWith('.faa')}
+                                        def gbk_found     = files.find{it.toString().tokenize('.').last().matches('gbk|gbff')}
+                                        def fasta         = fasta_found   != null ? fasta_found   : []
+                                        def faa           = faa_found     != null ? faa_found     : []
+                                        def gbk           = gbk_found     != null ? gbk_found     : []
```
Comment (Darcy220606) [inline workflows/funcscan.nf:92]: So this  PR also includes the addition of preannotated input right?

**Review by Darcy220606 — APPROVED** [review]
looks fine to me 👍🏽  Just this bit about the preannotated files as input in this PR...

File: workflows/funcscan.nf (hunk 3 above)
Comment (jfy133) [reply workflows/funcscan.nf:92]: Yes, will have to do both in one because I'm a dumdum and commited them and don't want to roll these back too..

**Comment by pentamorfico** [comment]
Hi! I saw that you included the input of pre-annotated ORFs in the new v2 release, but I cannot find any documentation on it.

Is it possible to use pre-annotated ORFs in GFF + FAA format? Would the output preserve the gene IDs based on the GFF file?

I'm running the workflow with the test data, and some of the output genes have the BAKTA ID, whereas some other have the Pyrodigal ID. Is there a way to use the same ID for all of them?

**Pull request merged**